File: string-paradigm-test-generator/src/test/java/org/jesperancinha/string/paradigm/inputgenerator/InputGeneratorTest.java
package org.jesperancinha.string.paradigm.inputgenerator;

import static com.google.common.truth.Truth.assertThat;

import java.io.InputStream;
import java.util.List;
import java.util.Stack;

import org.junit.Test;

public class InputGeneratorTest {

	private static final String TEST_FILMS_TXT = "testFilms.txt";

	@Test
	public void testGetFullFileList() throws Exception {
		final InputStream streamId = getClass().getResourceAsStream("sptg_id.txt");
		final InputStream streamActor = getClass().getResourceAsStream("sptg_actor.txt");
		final InputStream streamActress = getClass().getResourceAsStream("sptg_actress.txt");
		final InputStream streamDirector = getClass().getResourceAsStream("sptg_director.txt");
		final InputStream streamGenre = getClass().getResourceAsStream("sptg_genre.txt");
		final InputStream streamLocation = getClass().getResourceAsStream("sptg_location.txt");
		final InputStream streamMovie = getClass().getResourceAsStream("sptg_movie.txt");
		final InputStream streamProducer = getClass().getResourceAsStream("sptg_producer.txt");
		final InputStream streamSong = getClass().getResourceAsStream("sptg_song.txt");
		final InputStream streamStuntMan = getClass().getResourceAsStream("sptg_stunt_man.txt");
		final InputStream streamStuntWoman = getClass().getResourceAsStream("sptg_stunt_woman.txt");
		final InputStream streamYear = getClass().getResourceAsStream("sptg_year.txt");

		final InputGenerator inputGenerator = new InputGenerator();

		final List<String> resultId = inputGenerator.getFullFileList(streamId);
		final List<String> resultActor = inputGenerator.getFullFileList(streamActor);
		final List<String> resultActress = inputGenerator.getFullFileList(streamActress);
		final List<String> resultDirector = inputGenerator.getFullFileList(streamDirector);
		final List<String> resultGenre = inputGenerator.getFullFileList(streamGenre);
		final List<String> resultLocation = inputGenerator.getFullFileList(streamLocation);
		final List<String> resultMovie = inputGenerator.getFullFileList(streamMovie);
		final List<String> resultProducer = inputGenerator.getFullFileList(streamProducer);
		final List<String> resultSong = inputGenerator.getFullFileList(streamSong);
		final List<String> resultStuntMan = inputGenerator.getFullFileList(streamStuntMan);
		final List<String> resultStuntWoman = inputGenerator.getFullFileList(streamStuntWoman);
		final List<String> resultYear = inputGenerator.getFullFileList(streamYear);

		assertThat(resultId).hasSize(5);
		assertThat(resultActor).hasSize(8);
		assertThat(resultActress).hasSize(4);
		assertThat(resultDirector).hasSize(5);
		assertThat(resultGenre).hasSize(4);
		assertThat(resultLocation).hasSize(12);
		assertThat(resultMovie).hasSize(6);
		assertThat(resultProducer).hasSize(6);
		assertThat(resultSong).hasSize(7);
		assertThat(resultStuntMan).hasSize(6);
		assertThat(resultStuntWoman).hasSize(1);
		assertThat(resultYear).hasSize(12);
	}

	@Test
	public void testCreateTestFile() throws Exception {
		final InputStream streamId = getClass().getResourceAsStream("sptg_id.txt");
		final InputStream streamActor = getClass().getResourceAsStream("sptg_actor.txt");
		final InputStream streamActress = getClass().getResourceAsStream("sptg_actress.txt");
		final InputStream streamDirector = getClass().getResourceAsStream("sptg_director.txt");
		final InputStream streamGenre = getClass().getResourceAsStream("sptg_genre.txt");
		final InputStream streamLocation = getClass().getResourceAsStream("sptg_location.txt");
		final InputStream streamMovie = getClass().getResourceAsStream("sptg_movie.txt");
		final InputStream streamProducer = getClass().getResourceAsStream("sptg_producer.txt");
		final InputStream streamSong = getClass().getResourceAsStream("sptg_song.txt");
		final InputStream streamStuntMan = getClass().getResourceAsStream("sptg_stunt_man.txt");
		final InputStream streamStuntWoman = getClass().getResourceAsStream("sptg_stunt_woman.txt");
		final InputStream streamYear = getClass().getResourceAsStream("sptg_year.txt");

		Stack<InputStream> stack = new Stack<>();
		stack.push(streamSong);
		stack.push(streamStuntMan);
		stack.push(streamActor);
		stack.push(streamStuntWoman);
		stack.push(streamActress);
		stack.push(streamProducer);
		stack.push(streamDirector);
		stack.push(streamLocation);
		stack.push(streamGenre);
		stack.push(streamMovie);
		stack.push(streamYear);
		stack.push(streamId);

		final InputGeneratorOptions testOptions = new InputGeneratorOptions() {
			@Override
			public String getFilename() {
				return TEST_FILMS_TXT;
			}

			@Override
			public int getNumberOfParents() {
				return 10;
			}

			@Override
			public int getNumberOfElements() {
				return 2;
			}
		};

		final InputGenerator inputGenerator = new InputGenerator();
		inputGenerator.createTestFile(testOptions);

	}
}

File: string-paradigm-test-generator/src/test/kotlin/org/jesperancinha/string/paradigm/inputgenerator/InputGeneratorTest.kt
package org.jesperancinha.string.paradigm.inputgenerator

import io.kotest.matchers.collections.shouldHaveSize
import io.kotest.matchers.nulls.shouldNotBeNull
import org.junit.jupiter.api.Test
import java.io.InputStream
import java.util.*

class InputGeneratorTest {
    @Test
    @Throws(Exception::class)
    fun testGetFullFileList() {
        val streamId = javaClass.getResourceAsStream("sptg_id.txt").shouldNotBeNull()
        val streamActor = javaClass.getResourceAsStream("sptg_actor.txt".shouldNotBeNull())
        val streamActress = javaClass.getResourceAsStream("sptg_actress.txt").shouldNotBeNull()
        val streamDirector = javaClass.getResourceAsStream("sptg_director.txt").shouldNotBeNull()
        val streamGenre = javaClass.getResourceAsStream("sptg_genre.txt").shouldNotBeNull()
        val streamLocation = javaClass.getResourceAsStream("sptg_location.txt").shouldNotBeNull()
        val streamMovie = javaClass.getResourceAsStream("sptg_movie.txt").shouldNotBeNull()
        val streamProducer = javaClass.getResourceAsStream("sptg_producer.txt").shouldNotBeNull()
        val streamSong = javaClass.getResourceAsStream("sptg_song.txt").shouldNotBeNull()
        val streamStuntMan = javaClass.getResourceAsStream("sptg_stunt_man.txt").shouldNotBeNull()
        val streamStuntWoman = javaClass.getResourceAsStream("sptg_stunt_woman.txt").shouldNotBeNull()
        val streamYear = javaClass.getResourceAsStream("sptg_year.txt").shouldNotBeNull()
        val inputGenerator = InputGeneratorOptions()
        val resultId = inputGenerator.getFullFileList(streamId)
        val resultActor = inputGenerator.getFullFileList(streamActor)
        val resultActress = inputGenerator.getFullFileList(streamActress)
        val resultDirector = inputGenerator.getFullFileList(streamDirector)
        val resultGenre = inputGenerator.getFullFileList(streamGenre)
        val resultLocation = inputGenerator.getFullFileList(streamLocation)
        val resultMovie = inputGenerator.getFullFileList(streamMovie)
        val resultProducer = inputGenerator.getFullFileList(streamProducer)
        val resultSong = inputGenerator.getFullFileList(streamSong)
        val resultStuntMan = inputGenerator.getFullFileList(streamStuntMan)
        val resultStuntWoman = inputGenerator.getFullFileList(streamStuntWoman)
        val resultYear = inputGenerator.getFullFileList(streamYear)
        resultId.shouldHaveSize(5)
        resultActor.shouldHaveSize(8)
        resultActress.shouldHaveSize(4)
        resultDirector.shouldHaveSize(5)
        resultGenre.shouldHaveSize(4)
        resultLocation.shouldHaveSize(12)
        resultMovie.shouldHaveSize(6)
        resultProducer.shouldHaveSize(6)
        resultSong.shouldHaveSize(7)
        resultStuntMan.shouldHaveSize(6)
        resultStuntWoman.shouldHaveSize(1)
        resultYear.shouldHaveSize(12)
    }

    @Test
    @Throws(Exception::class)
    fun testCreateTestFile() {
        val streamId = javaClass.getResourceAsStream("sptg_id.txt")
        val streamActor = javaClass.getResourceAsStream("sptg_actor.txt")
        val streamActress = javaClass.getResourceAsStream("sptg_actress.txt")
        val streamDirector = javaClass.getResourceAsStream("sptg_director.txt")
        val streamGenre = javaClass.getResourceAsStream("sptg_genre.txt")
        val streamLocation = javaClass.getResourceAsStream("sptg_location.txt")
        val streamMovie = javaClass.getResourceAsStream("sptg_movie.txt")
        val streamProducer = javaClass.getResourceAsStream("sptg_producer.txt")
        val streamSong = javaClass.getResourceAsStream("sptg_song.txt")
        val streamStuntMan = javaClass.getResourceAsStream("sptg_stunt_man.txt")
        val streamStuntWoman = javaClass.getResourceAsStream("sptg_stunt_woman.txt")
        val streamYear = javaClass.getResourceAsStream("sptg_year.txt")
        val stack = Stack<InputStream>()
        stack.push(streamSong)
        stack.push(streamStuntMan)
        stack.push(streamActor)
        stack.push(streamStuntWoman)
        stack.push(streamActress)
        stack.push(streamProducer)
        stack.push(streamDirector)
        stack.push(streamLocation)
        stack.push(streamGenre)
        stack.push(streamMovie)
        stack.push(streamYear)
        stack.push(streamId)
        val inputGenerator: InputGeneratorOptions = object : InputGeneratorOptions() {
            override val filename: String
                get() = InputGeneratorTest.TEST_FILMS_TXT
            override val numberOfParents: Int
                get() = 10
            override val numberOfElements: Int
                get() = 2
        }
        inputGenerator.call()
    }

    companion object {
        private const val TEST_FILMS_TXT = "testFilms.txt"
    }
}